### PR TITLE
Host DataSource Added

### PR DESCRIPTION
**Pull request by Sakshi-dell**
# Description
Host Data source added

# GitHub Issues
List the GitHub issues impacted by this PR:

| GitHub Issue # |
| -------------- |
| |

# Checklist:

- [x] I have performed a self-review of my own code to ensure there are no formatting, vetting, linting, or security issues
- [x] I have verified that new and existing unit tests pass locally with my changes
- [x] I have not allowed coverage numbers to degenerate
- [ ] I have maintained at least 90% code coverage
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] Backward compatibility is not broken

# How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Please also list any relevant details for your test configuration

- [x] Acceptance tests
image

### CODE REVIEW

File: powerstore/datasource_host.go [hunk 1]
```diff
@@ -0,0 +1,367 @@
...
+										MarkdownDescription: "Username for CHAP authentication.",
+										Computed:            true,
+									},
+									"chap_single_username": schema.StringAttribute{
+										Description:         "Username for CHAP authentication.",
+										MarkdownDescription: "Username for CHAP authentication.",
+										Computed:            true,
+									},
+									"chap_single_password": schema.StringAttribute{
+										Description:         "Password for CHAP authentication.",
```
Comment (shenda1) [inline powerstore/datasource_host.go]: Are we getting password in the response?

File: go.mod [hunk 2]
```diff
@@ -3,7 +3,7 @@ module terraform-provider-powerstore
 go 1.19
 
 require (
-	github.com/dell/gopowerstore v1.11.1-0.20230320172746-beee61ee87f5
+	github.com/dell/gopowerstore v1.11.1-0.20230331110225-2e4fe0495575
 	github.com/hashicorp/terraform-plugin-docs v0.13.0
 	github.com/hashicorp/terraform-plugin-framework v1.0.1
```
Comment (shenda1) [inline go.mod]: Update the plugin framework version as discussed

File: powerstore/datasource_host.go (hunk 1 above)
Comment (Sakshi-dell) [reply powerstore/datasource_host.go]: Passwords were always blank in the response, removed passwords from the schema

File: go.mod (hunk 2 above)
Comment (Sakshi-dell) [reply go.mod]: Plugin framework will be taken up as a separate PR

**Review by shenda1 — DISMISSED** [review]

**Review by AnikaAgiwal2711 — DISMISSED** [review]

**Comment by AnikaAgiwal2711** [comment]
Looks good to me

**Review by shenda1 — APPROVED** [review]

**Review by AnikaAgiwal2711 — APPROVED** [review]

**Pull request merged**